File: Results/DER/Translation/translation_output/no_test/gpt-3.5/codenet/java/python/p00379.py
```python
import math

a, n, m = map(int, input().split())
max_val = int(math.pow(m, 1/n))

ans = 0
for i in range(1, max_val + 1):
    x = int(math.pow(i, n))
    str_x = str(x)
    y = sum(int(digit) for digit in str_x)
    
    if y + a == i:
        ans += 1

print(ans)
```

File: Results/DER/Translation/translation_output/no_test/codeqwen/codenet/python/java/p00379.java
Here is the Java version of your Python code:

```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int a = Integer.parseInt(input[0]);
        int n = Integer.parseInt(input[1]);
        int m = Integer.parseInt(input[2]);
        int ans = 0;
        for (int y = 1; y < 50; y++) {
            int x = (int) Math.pow(y + a, n);
            if (1 <= x && x <= m && y == sumDigits(x)) {
                ans += 1;
            }
        }
        System.out.println(ans);
    }

    public static int sumDigits(int n) {
        int sum = 0;
        while (n > 0) {
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }
}
```

This Java code does the same thing as your Python code. It reads three integers from the user, calculates the sum of digits of each number in the range 1 to 49 (inclusive), and counts how many of these numbers are equal to the sum of their digits. The result is then printed to the console.
